src/refresh.cc

```diff
@@ -727,6 +719,33 @@ refreshStats(StoreEntry * sentry)
         refreshCountsStats(sentry, refreshCounts[i]);
 }
 
+const RegexPattern &
+RefreshPattern::regex() const
+{
+    assert(regex_);
+    return *regex_;
+}
+
+void
+RefreshPattern::printPattern(std::ostream &os) const
+{
+    if (regex_)
+        regex_->print(os, nullptr); // refresh lines do not inherit line flags
+    else
+        os << "<none>";
```

Comment:
This function is used to dump the squid.conf settings. But I do not see anything in the new parser handling "<none>" as a special case pattern. 

I think this should be an assert(regex_) since refresh_pattern cannot be configured without a pattern and this function is for mgr:config dumping the squid.conf settings.

Reply:
My worry remains. Though I will trust you on this one and see what happens.